components/apiserver-proxy/cmd/proxy/main.go

```diff
@@ -354,3 +374,46 @@ func deleteUpstreamCORSHeaders(r *http.Response) error {
 	}
 	return nil
 }
+
+func setupReloadableOIDCAuthr(fileWatcherCtx context.Context, cfg *authn.OIDCConfig) (authenticator.Request, error) {
+	const eventBatchDelaySeconds = 10
+	filesToWatch := []string{cfg.CAFile}
+
+	//Create ReloadableAuthReq
+	authReqConstructorFunc := func() (authn.CancellableAuthRequest, error) {
+		glog.Infof("creating new instance of authenticator.Request...")
+		return authn.NewOIDCAuthenticator(cfg)
+	}
+
+	athntctr, err := r.NewReloadableAuthReq(authReqConstructorFunc)
```

Comment:
nit : Please change named import to reload or something else. It's confusing with othere methods/funcs  where r corresponds to Response